### DISCUSSION TITLE
How to enable Settings Sync in Codespaces automatically

### DISCUSSION THREAD

**Discussion by sagelga** [community/community#27269] [Codespaces] [answered]
When I get onto new web-based VSC Codespaces, my Settings Sync is always off by default.
I must enable it every time a new Codespace instance spins up. This is so annoying.

Can I do any settings or configurations to make my Codespaces enable a Settings Sync automatically after it spins up? Thanks!

**Comment by cwndrws** [comment, marked as answer]
Hi 👋 Thanks for your question!

When you create and connect to a new codespace in the web browser, a one-way sync is performed to pull all your settings into your new codespace once. If all you want is to ensure that your codespaces contain your settings, you don't need to enable settings sync at all. If you need to be able to push settings back up from a codespace or your settings have changed since you created the codespace and you want to sync the new settings in, you will need to enable the settings sync feature as you describe.

We don't automatically enable the full settings sync feature for security reasons. This is to prevent malicious code from pushing malicious extensions or settings into your settings somehow when you open it in a codespace. If you connect to your codespaces through the VS Code desktop editor, you can enable settings sync and it will persist to all codespaces you open from there.

**Reply by sagelga** [reply]
To be precise, do you mean **all of my settings** in Setting Sync is applied automatically to Browser editor once the Instance starts?

and you mentioned about “disable full settings sync by default”. Is there’s a way to turn it back on?

**Reply by cwndrws** [reply]
That's correct, you should get all your settings in a codespace once it starts. You can enable full settings sync by following this guide: https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-github-codespaces-for-your-account#settings-sync, but I think that's what you're already doing. There is not a way to persist that setting for codespaces in the web. You would have to do this for every codespace you create or use VS Code desktop.

**Reply by sagelga** [reply]
Whoopsies! I mean how can I *enable full settings sync* automatically or by default. 
I currently have to turn it on manually every time the new instance starts on Web Editor.

btw, thanks for the answer!
